File: dev-docs/snippets/rule-sets/schema.swift
import JsBaoClient

// Retrieve the rule set schema. Swift returns a typed `RuleSetSchema`
// (`resourceTypes` is an opaque `[String: JSONValue]` map).
func schema(client: JsBaoClient) async throws {
  // #region example
  let schema = try await client.ruleSets.schema()
  // #endregion example
  _ = schema
}

File: dev-docs/snippets/rule-sets/schema.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Retrieve the rule set schema describing available resource types.
export async function schema(client: JsBaoClient) {
  // #region example
  const schema = await client.ruleSets.schema();
  // #endregion example
  return schema;
}
